src/proto/mgmt/pool.proto

```diff
@@ -58,17 +58,17 @@ message PoolDestroyResp {
 	int32 status = 1; // DAOS error code
 }
 
+// ListPoolsReq represents a request to list pools on a given DAOS system.
 message ListPoolsReq {
 	string sys = 1; // DAOS system identifier
-	uint64 numPools = 2; // Client response buffer capacity in number of pools
 }
 
+// ListPoolsResp returns the list of pools in the system.
 message ListPoolsResp {
 	message Pool {
 		string uuid = 1; // uuid of pool
-		string svcreps = 2; // pool service replicas, comma-separated integers
+		repeated int32 svcreps = 2; // pool service replica ranks
 	}
 	int32 status = 1; // DAOS error code
-	repeated Pool pools = 2; // pools list (max length ListPoolsReq.numPools)
-	uint64 numPools = 3; // number of pools in system
```

Comment:
In Go we don't preallocate a buffer for the Protobuf response, so the whole concept of requesting the length, allocating a buffer, and then sending a separate RPC to request the content doesn't really make sense.

Things could get nasty in the dRPC protocol itself - we have a message max size so it may become necessary to chunk content at that point if it's too big. But I'm not sure how knowing the total number of pools or containers would fit into that - we'd want to be able to ask for a specific subset at a time. Not sure how we should approach that, but I think it's good to start thinking about.